overlord/devicestate/devicestate.go

```diff
@@ -1358,9 +1378,23 @@ type recoverySystemSetup struct {
 	// are kept, typically /run/mnt/ubuntu-seed/systems/<label>, set when
 	// tasks are created
 	Directory string `json:"directory"`
-	// SnapSetupTasks is a list of task IDs that carry snap setup
-	// information, relevant only during remodel, set when tasks are created
+	// SnapSetupTasks is a list of task IDs that carry snap setup information.
+	// Tasks could come from a remodel, or from downloading snaps that were
+	// required by a validation set.
 	SnapSetupTasks []string `json:"snap-setup-tasks"`
+	// LocalSnaps is a list of snap.SideInfo structs that represent snaps thtat
+	// should be used to create the recovery system.
+	LocalSnaps []*snap.SideInfo `json:"local-snaps,omitempty"`
+	// LocalSnapPaths is a list of paths to snaps that corresponds to the list
+	// of snap.SideInfo structs in SideInfos.
+	LocalSnapPaths []string `json:"local-snap-paths,omitempty"`
+	// TestSystem is set to true if the new recovery system should
+	// not be verified by rebooting into the new system. Once the system is
+	// created, it will immediately be considered a valid recovery system.
+	TestSystem bool `json:"test-system,omitempty"`
```

Comment:
Do I need to worry about this being marshaled to false in the case that it isn't there? Like, if the task was created prior to an update of snapd. I'm not sure if that is possible. For safety, I also included checks to always test the system during a remodel.

Reply:
that might be safer as a remodel could update snapd itself?  another approach would be to add some kind of versioning to this struct

Reply:
Made and tested this change in da2a8465c388c787f9f2c129d15b182fcd5958fe.